src/frontend/src/pages/TeamsPage/TeamMembersPageBlock.tsx

```diff
@@ -0,0 +1,129 @@
+    );
+  };
+  const userToAutocompleteOption = (user: User): { label: string; id: number } => {
+    return { label: `${fullNamePipe(user)}`, id: user.userId };
+  };
+
+  return isEditingMembers ? (
+    <PageBlock
+      title={'People'}
+      headerRight={
+        <div style={{ display: 'flex' }}>
+          <Button onClick={() => setIsEditingMembers(false)}>Cancel</Button>
+          <Button
+            sx={{
+              ml: 2,
+              backgroundColor: theme.palette.success.main,
+              color: theme.palette.success.contrastText,
+              '&:hover': {
+                backgroundColor: theme.palette.success.dark
+              }
+            }}
+            onClick={handleSubmit}
+          >
+            Save
+          </Button>
+        </div>
+      }
+    >
+      <Grid container spacing={1}>
+        <Grid item xs={12}>
+          <Typography sx={{ display: 'inline', fontWeight: 'bold' }}>Lead: </Typography>
+          <Typography sx={{ display: 'inline' }} variant={'body1'}>
+            {fullNamePipe(team.leader)}
+          </Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Autocomplete
+            multiple
+            id="tags-standard"
+            //first filters the options to only include users who have not been selected to be on the team, then sorts the options by alphabetical order of their first name
+            options={data
+              .filter((user) => !memberIds.includes(user.userId))
+              .sort((a, b) => (a.firstName > b.firstName ? 1 : -1))
+              .map(userToAutocompleteOption)}
+            defaultValue={team.members.map(userToAutocompleteOption)}
+            onChange={(event, newValue) => {
+              setMemberIds(newValue.map((option) => option.id));
+            }}
+            isOptionEqualToValue={(option, value) => option.id === value.id}
+            renderInput={(params) => (
+              <TextField {...params} variant="standard" label="Members" placeholder="Select A User" />
+            )}
+          />
+        </Grid>
+      </Grid>
+    </PageBlock>
+  ) : (
+    <PageBlock
+      title={'People'}
+      headerRight={<IconButton disabled={hasPerms()} onClick={() => setIsEditingMembers(true)} children={<Edit />} />}
+    >
+      <Grid container spacing={1}>
+        <Grid item xs={12}>
+          <Typography style={{ display: 'inline', fontWeight: 'bold' }}>Lead: </Typography>
+          <Typography style={{ display: 'inline' }}>{fullNamePipe(team.leader)}</Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Typography sx={{ display: 'inline', fontWeight: 'bold' }}>Members: </Typography>
+          <Typography sx={{ display: 'inline' }}>
+            {team.members
+              .map((member) => {
+                return fullNamePipe(member);
+              })
+              .join(', ')}
+          </Typography>
+        </Grid>
+      </Grid>
+    </PageBlock>
```

Comment:
ohhh I see it's a ternary, I would put each big component in a helper, so it's more clear what's going on with the conditional rendering.